---a/net/minecraft/item/ItemFireball.java
+++b/net/minecraft/item/ItemFireball.java
@@ -8,39 +8,38 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
-public class ItemFireball extends Item
-{
+public class ItemFireball extends Item {
+
     private static final String __OBFID = "CL_00000029";
 
-    public ItemFireball()
-    {
+    public ItemFireball() {
         this.setCreativeTab(CreativeTabs.tabMisc);
     }
 
-    public boolean onItemUse(ItemStack stack, EntityPlayer playerIn, World worldIn, BlockPos pos, EnumFacing side, float hitX, float hitY, float hitZ)
-    {
-        if (worldIn.isRemote)
-        {
+    public boolean onItemUse(ItemStack stack, EntityPlayer playerIn, World worldIn, BlockPos pos, EnumFacing side, float hitX, float hitY, float hitZ) {
+        if (worldIn.isRemote) {
             return true;
-        }
-        else
-        {
+        } else {
             pos = pos.offset(side);
 
-            if (!playerIn.canPlayerEdit(pos, side, stack))
-            {
+            if (!playerIn.canPlayerEdit(pos, side, stack)) {
                 return false;
-            }
-            else
-            {
-                if (worldIn.getBlockState(pos).getBlock().getMaterial() == Material.air)
-                {
-                    worldIn.playSoundEffect((double)pos.getX() + 0.5D, (double)pos.getY() + 0.5D, (double)pos.getZ() + 0.5D, "item.fireCharge.use", 1.0F, (itemRand.nextFloat() - itemRand.nextFloat()) * 0.2F + 1.0F);
+            } else {
+                if (worldIn.getBlockState(pos).getBlock().getMaterial() == Material.air) {
+                    // CraftBukkit start - fire BlockIgniteEvent
+                    if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockIgniteEvent(worldIn, pos.getX(), pos.getY(), pos.getZ(), org.bukkit.event.block.BlockIgniteEvent.IgniteCause.FIREBALL, playerIn).isCancelled()) {
+                        if (!playerIn.capabilities.isCreativeMode) {
+                            --stack.stackSize;
+                        }
+                        return false;
+                    }
+                    // CraftBukkit end
+                    worldIn.playSoundEffect((double) pos.getX() + 0.5D, (double) pos.getY() + 0.5D, (double) pos.getZ() + 0.5D, "item.fireCharge.use", 1.0F, (itemRand.nextFloat() - itemRand.nextFloat()) * 0.2F + 1.0F);
                     worldIn.setBlockState(pos, Blocks.fire.getDefaultState());
                 }
 
-                if (!playerIn.capabilities.isCreativeMode)
-                {
+                if (!playerIn.capabilities.isCreativeMode) {
                     --stack.stackSize;
                 }
 
@@ -48,4 +48,4 @@
             }
         }
     }
-}
+}
